src/main/java/net/openhft/chronicle/core/internal/threads/VanillaThreadLock.java

```diff
@@ -0,0 +1,163 @@
+package net.openhft.chronicle.core.internal.threads;
+
+import net.openhft.chronicle.core.Jvm;
+import net.openhft.chronicle.core.OS;
+import net.openhft.chronicle.core.threads.InterruptedRuntimeException;
+import net.openhft.chronicle.core.threads.ThreadLock;
+import net.openhft.chronicle.core.values.LongValue;
+
+import java.io.File;
+
+public class VanillaThreadLock implements ThreadLock {
+    static final Metrics METRICS = new Metrics();
+    private static final long TOP_BITS = ~0L << 32;
+    private static final int UNLOCKED = 0;
+    private final LongValue twoThreadId;
+    private final long timeoutMs;
+    int busyLoopCount = 20_000;
+    int busyLockSlowerCount = 100;
+    private long oldLock;
+
+    public VanillaThreadLock(LongValue twoThreadId, long timeoutMs) {
+        this.twoThreadId = twoThreadId;
+        this.timeoutMs = timeoutMs;
+    }
+
+    @Override
+    public boolean tryLock(int threadId) throws IllegalStateException {
+        if (threadId == 0)
+            throw new IllegalArgumentException();
+        long twoThreadId0 = twoThreadId.getVolatileValue();
+        int threadId0 = (int) twoThreadId0;
+        if (threadId0 != UNLOCKED) {
+            if (threadId0 == threadId)
+                throw new IllegalStateException("trying to lock twice");
+            return false;
+        }
+        return twoThreadId.compareAndSwapValue(twoThreadId0, (twoThreadId0 & TOP_BITS) | threadId);
+    }
+
+    @Override
+    public void lock(int threadId) throws InterruptedRuntimeException {
+        if (tryLock(threadId))
+            return;
+        busyLock(threadId);
+    }
+
+    private void busyLock(int threadId) {
+        // busy wait
+        for (int i = 0; i < busyLoopCount; i++) {
+            if (tryLock(threadId))
+                return;
+            Jvm.nanoPause();
+        }
+        if (busyLockSlower(threadId))
+            return;
+        forceUnlockAndRetry(threadId);
+    }
+
+    @SuppressWarnings("java:S3776")
```

Comment:
Rather than suppressing the warning for cognitive overload, please consider breaking up the logic into steps.